cajita/src/Cajita_ParameterPack.hpp

```diff
@@ -12,169 +12,45 @@
 #ifndef CAJITA_PARAMETERPACK_HPP
 #define CAJITA_PARAMETERPACK_HPP
 
-#include <Kokkos_Core.hpp>
-
-#include <type_traits>
-#include <utility>
+#include <Cabana_Core.hpp>
 
 namespace Cajita
 {
-//---------------------------------------------------------------------------//
-// Parameter pack device capture.
-//
-// NOTE: In general this would not be needed but NVCC cannot capture parameter
-// packs in lambda functions hence we need to wrap them in something that can
-// be captured.
-//---------------------------------------------------------------------------//
-// Get the type at the given index of a paremeter pack.
-template <std::size_t N, typename T, typename... Types>
-struct PackTypeAtIndexImpl;
-
-template <typename T, typename... Types>
-struct PackTypeAtIndexImpl<0, T, Types...>
-{
-    using type = T;
-};
-
-template <std::size_t N, typename T, typename... Types>
-struct PackTypeAtIndexImpl
-{
-    using type = typename PackTypeAtIndexImpl<N - 1, Types...>::type;
-};
-
-template <std::size_t N, typename... Types>
-struct PackTypeAtIndex
-{
-    using type = typename PackTypeAtIndexImpl<N, Types...>::type;
-    static_assert( N < sizeof...( Types ), "Type index out of bounds" );
-};
-
-//---------------------------------------------------------------------------//
-// Parameter pack element.
-template <std::size_t N, typename T>
-struct ParameterPackElement
-{
-    T _m;
-};
-
-//---------------------------------------------------------------------------//
-// Capture a parameter pack. All parameter pack elements must be copyable to
-// device.
-template <typename Sequence, typename... Types>
-struct ParameterPackImpl;
-
-template <std::size_t... Indices, typename... Types>
-struct ParameterPackImpl<std::index_sequence<Indices...>, Types...>
-    : ParameterPackElement<Indices, Types>...
-{
-};
-
-template <typename... Types>
-struct ParameterPack
-    : ParameterPackImpl<std::make_index_sequence<sizeof...( Types )>, Types...>
-{
-    template <std::size_t N>
-    using value_type = typename PackTypeAtIndex<N, Types...>::type;
-
-    template <std::size_t N>
-    using const_value_type = typename std::add_const<value_type<N>>::type;
-
-    template <std::size_t N>
-    using element_type = ParameterPackElement<N, value_type<N>>;
-
-    static constexpr std::size_t size = sizeof...( Types );
-};
-
-//---------------------------------------------------------------------------//
 // Static type checker.
-template <class>
-struct is_parameter_pack_impl : public std::false_type
+template <class T>
+struct is_parameter_pack : public Cabana::is_parameter_pack_impl<
+                               typename std::remove_cv<T>::type>::type
 {
 };
 
+// Create a parameter pack.
 template <typename... Types>
-struct is_parameter_pack_impl<ParameterPack<Types...>> : public std::true_type
+Cabana::ParameterPack<Types...> makeParameterPack( const Types&... ts )
```

Comment:
Why duplicate this function here?

Reply:
It was only for full backwards compatibility. I can remove